regression/contracts/loop_assigns-05/main.c

```diff
@@ -0,0 +1,38 @@
+#include <assert.h>
+
+int j;
+
+int lowerbound()
+{
+  return 0;
+}
+
+int upperbound()
+{
+  return 10;
+}
+
+void incr(int *i)
+{
+  (*i)++;
+}
+
+void body(int i)
+{
+  j = i;
+}
+
+void main()
+{
+  for(int i = lowerbound(); i < upperbound(); incr(&i))
```

Comment:
Agreed that you should be able to find these and agreed that this would be worth adding as a test.
Not entirely sure how this connects to the specific issue being fixed though.